case-lib/lib.sh

```diff
@@ -752,17 +752,84 @@ func_lib_check_pa()
 # However, 1. arrays would complicate the user interface 2. ALSA does not
 # seem to need arguments with whitespace or globbing characters.
 
+# SOF_ALSA_TOOL:
+# This option is used for selecting tool for testing,
+# So far, supported tools are 'alsa' and 'tinyalsa'
+# To select appropriate tool, set SOF_ALSA_TOOL to one of above
+# before using 'aplay_opts' or 'arecord_opts' function.
+# (e.g., SOF_ALSA_TOOL=alsa)
+
+
+# Function to extract the card number and device number from $dev option (e.g., hw:0,10)
+parse_audio_device() {
+    # Extract the card number (e.g., "0" from hw:0,10)
+    card_nr=$(printf '%s' "$dev" | cut -d ':' -f2 | cut -d ',' -f1)
```

Comment:
Avoiding global variables is sometimes difficult in shell but not in this case. In this case there's no reason to use a global variable.

```suggestion
    card_nr=$(printf '%s' "$1" | cut -d ':' -f2 | cut -d ',' -f1)
```